frontend/apps/filemanager/filemanagerbookinfo.lua

```diff
@@ -119,18 +144,31 @@ function BookInfo:show(file, book_props)
     end
 
     local KeyValuePage = require("ui/widget/keyvaluepage")
-    local widget = KeyValuePage:new{
+    self.kvp_widget = KeyValuePage:new{
         title = _("Book information"),
         value_overflow_align = "right",
         kv_pairs = kv_pairs,
         values_lang = values_lang,
+        close_callback = function()
+            if self.updated then
+                if caller_callback then
+                    caller_callback()
+                end
+                local FileManager = require("apps/filemanager/filemanager")
+                if FileManager.instance then
+                    FileManager.instance:onRefresh()
+                end
+            end
+        end,
     }
-    UIManager:show(widget)
+    UIManager:show(self.kvp_widget)
 end
 
 function BookInfo:getBookProps(file, book_props, no_open_document)
+    local cover
     if DocSettings:hasSidecarFile(file) then
         local doc_settings = DocSettings:open(file)
+        cover = doc_settings:readSetting("doc_cover")
```

Comment:
It seems this `doc_cover` is never written to ?

Reply:
Thanks, leftover.